File: Release/MatlabFunctions/ManipulateTXTFile.m
% Function: ManipulateTXTFile String replacement in .txt file
% -----------------------------
% Usage:
% n = ManipulateTXTFile(TXTFile,StringToReplace,NewString)
% -----------------------------
% Input:
% TXTFile           String with name of .txt file
% StringToReplace   String which has to be replaced
% NewString         String with replacement
% -----------------------------
% Output:
% n                 number of replacements
% -----------------------------
% Created: 
% David Schlipf on 05-Dec-2010
% (c) Universitaet Stuttgart and sowento GmbH
% ----------------------------------
function n = ManipulateTXTFile(TXTFile,StringToReplace,NewString)

[FOLDER,NAME,EXT]   = fileparts(TXTFile); 
TempTXTFile         = fullfile(FOLDER,[NAME,'_temp',EXT]);
fid                 = fopen(TXTFile);
fidTemp             = fopen(TempTXTFile,'w+');
n                   = 0;

while ~feof(fid)
   s            = fgetl(fid);
   sTemp        = strrep(s,StringToReplace,NewString);
   fprintf(fidTemp,'%s\r\n',sTemp);
   if ~strcmp(s,sTemp)
       n = n+1;
   end
end

fclose(fid);
fclose(fidTemp);
delete(TXTFile);
movefile(TempTXTFile,TXTFile);

File: Release/PythonFunctions/FileOperations/ManipulateTXTFile.py
import os


def ManipulateTXTFile(txt_file, string_to_replace, new_string):
    folder, name_ext = os.path.split(txt_file)
    name, ext = os.path.splitext(name_ext)
    temp_txt_file = os.path.join(folder, name + '_temp' + ext)
    n = 0

    with open(txt_file, 'r') as fid, open(temp_txt_file, 'w') as fid_temp:
        for line in fid:
            s = line.strip()
            s_temp = s.replace(string_to_replace, new_string)
            fid_temp.write(s_temp + '\n')
            if s != s_temp:
                n += 1

    os.remove(txt_file)
    os.rename(temp_txt_file, txt_file)

    return n
# source: Matlab-Function (ManipulateTXTFile.m)